### PR TITLE
add getNonceLock tests

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

* Add scenario to `getNonceLock` to checks if the right nonceTracker is used when a networkClientId is provided
* Add integration tests for `getNonceLock` that test result and that an acquired lock blocks subsequent attempts to acquire
* Lint IncomingTxOptions jsdoc 

### `@metamask/package-a`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

### `@metamask/package-b`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by shanejonas — APPROVED** [review]

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 1]
```diff
@@ -1721,4 +1721,259 @@ describe('TransactionController Integration', () => {
...
+          },
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:1747]: Do we need two of the same mocks here? If so why?

File: packages/transaction-controller/src/TransactionControllerIntegration.test.ts [hunk 2]
```diff
@@ -1721,4 +1721,259 @@ describe('TransactionController Integration', () => {
...
+              },
+            },
+            // NonceTracker
+            {
+              request: {
+                method: 'eth_getTransactionCount',
+                params: [ACCOUNT_MOCK, '0x1'],
+              },
+              response: {
+                result: '0xa',
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionControllerIntegration.test.ts:1782]: [nit] should we mock a different result for different networkClients so we can test more precisely?

Comment (adonesky1) [reply packages/transaction-controller/src/TransactionControllerIntegration.test.ts:1782]: oh i see the assertion is in the for loop still

**Review by adonesky1 — APPROVED** [review]
nice work!

**Pull request merged**